src/overseers/overseer.py

```diff
@@ -66,33 +88,55 @@ def mytardis_setup(self) -> IntrospectionConfig:
         """Getter for mytardis_setup. Sends API request if self._mytardis_setup is None"""
         return self._mytardis_setup or self.get_mytardis_setup()
 
-    @staticmethod
-    def resource_uri_to_id(uri: URI) -> int:
-        """Gets the id from a resource URI
+    def type_supports_identifiers(self, object_type: MyTardisObject) -> bool:
+        """Check if the given object type supports identifiers"""
+        if self.mytardis_setup.objects_with_ids is None:
+            return False
+        # pylint: disable=unsupported-membership-test
+        return object_type in self.mytardis_setup.objects_with_ids
 
-        Takes resource URI like: http://example.org/api/v1/experiment/998
-        and returns just the id value (998).
+    def generate_identifier_matchers(
+        self, object_type: MyTardisObject, object_data: dict[str, Any]
+    ) -> Generator[dict[str, Any], None, None]:
+        """Generate object matchers for identifiers"""
 
-        Args:
-            uri: str - the URI from MyTardis
+        if self.mytardis_setup.objects_with_ids is None:
+            raise ValueError("The MyTardis instance does not support identifier search")
 
-        Returns:
-            The integer id that maps to the URI
-        """
-        uri_sep: str = "/"
-        return int(urlparse(uri).path.rstrip(uri_sep).split(uri_sep).pop())
+        # pylint: disable=unsupported-membership-test
+        if object_type not in self.mytardis_setup.objects_with_ids:
+            raise ValueError(
+                f"MyTardis object type '{object_type}' does not support identifier search"
+            )
+
+        if identifiers := object_data.get("identifiers"):
+            for identifier in identifiers:
+                yield {"identifier": identifier}
+
+        # NOTE: if there are no identifiers, we'll yield nothing - is that an error case?
```

Comment:
It doesn't seem to cause any issues downstream and identifiers are optional so it feels like the intended outcome to generate nothing if iterating over an missing list.

Reply:
Or did you mean to add an error elsewhere if nothing can be found based on the uids and the identifiers?

Reply:
Good catch, I agree it's probably the right outcome so I removed the comment